### DISCUSSION TITLE
<newbie>recursing into eldoc

### DISCUSSION THREAD

**Discussion by yatesco** [joaotavora/eglot#1316] [Q&A] [answered]
(this isn't strictly about eglot, so apologies, but it is a pain point I'm hitting repeatedly trying to use Eglot)
(newbie, so I'm not even sure if I'm asking the correct question!)

If I open up the documentation (`C-h .`) for a symbol then I get an `eldoc` buffer:

<img width="768" alt="image" src="https://github.com/joaotavora/eglot/assets/181469/7d62bb29-c31e-4fce-8912-d979d60a16e2">

I `C-x o` over to it and scroll around and see a thing I want to navigate to:

<img width="768" alt="image" src="https://github.com/joaotavora/eglot/assets/181469/6ace2ded-2699-4d45-ad72-f8567f0ab845">

What I'd like to do is replace the current `eldoc` buffer with the result of doing `C-h .` in the "main" buffer on the same `type`, but do `C-h .` *within* the `eldoc` buffer:

<img width="768" alt="image" src="https://github.com/joaotavora/eglot/assets/181469/6952058d-f038-4edc-93a1-57982a2ac7d8">

but nothing I do in the `eldoc` buffer seems to "navigate" through to the symbol at point. Is this possible?

(to achieve the above image I had to create a variable of the same `type` in the main buffer and then `C-h .` on it. This is obviously not scalable ;-))

Thanks!

**Comment by joaotavora** [comment, marked as answer]
Yup, this doesn't work.  For one, ElDoc and the `*eldoc*` buffer is an at-point documentation system, meaning it is mostly designed to show documentation _alongside_ the source code, based on the position of point (which is what the cursor is called in Emacs).

Secondly, even if ElDoc were enhanced to allow navigation within the `*eldoc*` buffer (which would be possible but not necessarily a good idea, at least in those exact terms)  LSP, the protocol that Eglot uses to collect documentation, doesn't have any interface for asking documentation about a particular symbol or following hyperlinks.

However, as you have found out, some types of documentation rendered in the `*eldoc*` buffer contains hyperlinks that may or may not "work".  This depends on the renderer.  It could be that some of them open in your web browser, which might even be Emacs's own web browser, eww.    So is this really what you asked for?  Not really, it depends on the LSP server providing the documentation, and on the hyperlink meta-information it sends along with it.  It also depends on the renderer being used.

**Reply by yatesco** [reply]
I guess as well, the `*eldoc*` buffer isn't _parsing_ the text it's presenting, so it can't distinguish a code fragment (on which I'd like to `C-h .`) from any other text...

What's the general answer here? How do you solve this problem personally? I generally want either my code (which LSP can provide) or the general library, so is there a way of telling `xref` to include the `std` library?

(thanks again for your support!)
